components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -229,6 +231,7 @@ func newWheeledOdometry(
 		lastLeftPos:  0.0,
 		lastRightPos: 0.0,
 		originCoord:  geo.NewPoint(0, 0),
+		coord:        geo.NewPoint(0, 0),
```

Comment:
Mmmm, we actually didn't initialize `o.coord` because nil positions end up as NaNs if they are nil, semantically differentiating them from a GPS concept called "null island". Could you remove the setting of `o.coord` in the constructor?

Reply:
[tangent] I wouldn't really call [Null Island](https://en.wikipedia.org/wiki/Null_Island) a GPS-specific concept, more of a  mapping one.

Regardless; these do not end up as NaNs. I believe a failure to set `o.coord` is strictly incorrect. Here's why:

As `Position` returns a `*geo.Point`, a failure to set `coord` returns a nil pointer, not a NaN position. The end of `Reconfigure()` initializes `trackPosition()`, which will eventually set `o.coord` after some wait time, based on the relative position from `o.originCoord`. However, this introduces a race condition; if `Position()` is called before the first iteration of `trackPosition()` completes, the user will receive a `nil`, while if they had waited a moment longer they would receive a valid `*geo.Point` whose location is tightly tied to `o.originCoord`.

This race condition is also why we *must* set `o.coord` in the `DoCommand`. Executing a `DoCommand` to change `o.originCoord` without also updating `o.coord` again creates a race condition, where if `Position()` is called before the next iteration of `trackPosition()` completes, the user will receive stale data.

Reply:
[reply to tangent] gps systems do need  null island <-> error bad data sanitizing even in the case of just polling and storing GPS data from a stationary sysem, so I would call null island a GPS concept.

We do force the `nil` returned from the `Position` method to NaNs in our most recent movement sensor client code, if that was snot happening for you, please file a bug ticket. This is to account for  bad reader or calculation outputs from modules written for movement sensors that we do not own. 

I do want a badly calculated position or a nil coordinate to mean something that is different than the 0 lat 0 long coordinate. I am happy to initialize it as `geo.Point(math.Nan,math.NaN)` to keep  that behaviour consistent and deal with the race with Reocnfigure, which I would expect to output a nil in the case you described.

Reply:
[tangent nit] IMO that would make it a concept that GPS makes use of, much like general and special relativity, not a GPS (implied GPS-specific) concept :)

Forcing a `nil` to NaNs in the client is great, but in this case what the other RDK tests are using is what is returned by the constructor. I can handle this in my consuming code.

However, I would push back on initializing `o.coord` differently than `o.originCoord`. By initializing `o.originCoord`, that is saying that we will be computing positions relative to that position. That is the starting point. If the race condition is hit and a movement away from our starting point has not yet been calculated, then we are at our starting point. Similarly, if we use a DoCommand to change our origin coordinate/starting point, that is describing a known location where we are.

As it currently stands, the current behavior is instead "Return erroneous/bad data for an arbitrary/unknown amount of time, then eventually return the position that was known all along".

Reply:
We could initialize both as `geo.Point(math.Nan,math.NaN)`, but would make `wheeledodometry` somewhat useless until a DoCommand is called, which is a worse user experience than the current one where the bug is on a timer and resolves itself on its own.

Reply:
I guess I'm having trouble seeing the described nil/NaN behavior in specifically `wheeledodometry` as anything other than a bug. Could you help me understand and describe a scenario where that would instead be desirable?